tools/crm_rule.c

```diff
@@ -135,22 +135,22 @@ crm_rule_check(pe_working_set_t *data_set, const char *rule_id, crm_time_t *effe
 
     if (result == pe_date_within_range) {
         printf("Rule %s is still in effect\n", rule_id);
-        rc = 0;
+        rc = pcmk_rc_ok;
     } else if (result == pe_date_op_satisfied) {
         printf("Rule %s satisfies conditions\n", rule_id);
-        rc = 0;
+        rc = pcmk_rc_ok;
     } else if (result == pe_date_after_range) {
         printf("Rule %s is expired\n", rule_id);
-        rc = 1;
+        rc = -1;
```

Comment:
While this would work, it's treading on shaky ground. Negative values should be presumed reserved, whether for pacemaker (<= -1001) or for external callers. We could either add new pcmk_rc_ values, or have the function return crm_exit_t directly. Since pe_eval_date_result_t is internal, we could even potentially replace it with pcmk_rc_ entirely.

Reply:
Oh, I thought the purpose of the -1 to -1000 reserved space was that callers could use whatever values in that range as a kind of one-off set of return values.  I guess of those options, I'd prefer to merge pe_eval_date_result_t into pcmk_rc_e.  I'd like to not have functions return crm_exit_t if I can help it.